addons/storysource/src/StoryPanel.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import { API } from '@storybook/api';
+import { styled } from '@storybook/theming';
+import { Link } from '@storybook/router';
+import {
+  SyntaxHighlighter,
+  SyntaxHighlighterProps,
+  SyntaxHighlighterRendererProps,
+} from '@storybook/components';
+
+import createElement from 'react-syntax-highlighter/dist/esm/create-element';
+
+const StyledStoryLink = styled(Link)<{ to: string; key: string }>(({ theme }) => ({
+  display: 'block',
+  textDecoration: 'none',
+  borderRadius: theme.appBorderRadius,
+  color: 'inherit',
+
+  '&:hover': {
+    background: theme.background.hoverable,
+  },
+}));
+
+const SelectedStoryHighlight = styled.div(({ theme }) => ({
+  background: theme.background.hoverable,
+  borderRadius: theme.appBorderRadius,
+}));
+
+const StyledSyntaxHighlighter = styled(SyntaxHighlighter)<SyntaxHighlighterProps>(({ theme }) => ({
+  fontSize: theme.typography.size.s2 - 1,
+}));
+
+interface SourceLoc {
```

Comment:
can you pull these types out into a file, maybe as part of `source-loader` so that they can be reused between here & `addon-docs`?

Reply:
thanks, fixing. Also re-arranging the source-loader code to remove the `server` folder and have the code in src, since now there is no `preview` at all